packages/sentry/index.tsx

```diff
@@ -30,14 +59,26 @@ export const sentryFetch: typeof fetch = async (resource, options) => {
             console.error(msg, context);
          }
          return response;
-      })
-      .catch((error) => {
+      } catch (error) {
          // We don't want to hear about network errors in Sentry
          console.error(error, context);
          throw error;
-      });
+      }
+   };
+
+const shouldSkipReporting: SkipRequestFn = (input, init) => {
+   const url = getRequestUrl(input);
+   // We have custom logic in place for reporting errors only after React Query retries have failed,
+   // so we don't want to report errors for the cart API here.
+   return url.includes('/store/user/cart');
```

Comment:
I think I'd be more inclined to have our overridden `fetch()` take a `reportErrors: false` in it's `init` options instead of a regex on urls. I think the explicitness of that choice in the calling code makes it easier to understand.